source/open_node/ast_visitor/ast_visitor_examples.py

```diff
@@ -0,0 +1,71 @@
+from open_node.ast import (
+    Identifier,
+    QuantumGateModifier,
+    QubitDeclaration,
+    QuantumGate,
+    QuantumGateDefinition,
+)
+from open_node.ast_visitor.ast_visitor import NodeVisitor, NodeTransformer
+
+
+class OpenNodePrinter(NodeVisitor):
+    """
+    Example visitor which "unparses" the Program back to text
+    """
+
+    def __init__(self):
+        self.out = ""
+
+    def _print(self, line):
+        self.out += line
+        self.out += "\n"
+
+    def visit_QuantumGateDefinition(self, node: QuantumGateDefinition):
+        self._print(f"gate {node.name} {', '.join(q.name for q in node.qubits)}" " {")
+        for line in node.body:
+            self.out += "    "
+            self.visit(line)
+        self._print("}\n")
+
+    def visit_QubitDeclaration(self, node: QubitDeclaration):
+        self._print(f"qubit {node.qubit.name};")
+
+    def visit_QuantumGate(self, node: QuantumGate):
+        for modifier in node.modifiers:
+            self._print(f"{self.visit(modifier)} @ ")
+        self._print(f"{node.name} {', '.join(q.name for q in node.qubits)};")
+
+    def visit_QuantumGateModifier(self, node: QuantumGateModifier):
+        self._print(node.modifier)
+        if node.argument:
+            self._print(f"({self.visit(node.argument)})")
+
+
+class GateResolver(NodeTransformer):
```

Comment:
This example can be removed as well since it has not Contextual Analysis, meaning that it cannot correctly replace the logical qubit name when resolving a gate.

Reply:
I removed ast_visitor_examples.py as you suggested in this and one of the previous comments.